### PR TITLE
Feat/wallet with context

### PR DESCRIPTION
**Pull request by luckyOdisetti**
* Move wallet hook into context
* Disable WalletConnect support
* Fix memory leak and unnecessary re-renders on wallet
* Metamask network change now reloads all data in the app without a refresh
* Remove network passed via params and read from context
* Format pair name

### CODE REVIEW

File: packages/client/src/components/connect-wallet-button.tsx [hunk 1]
```diff
@@ -1,13 +1,12 @@
 import config from 'config';
-import { Wallet } from 'types/states';
+import {useWallet} from 'hooks/use-wallet';
```
Comment (kkennis) [inline packages/client/src/components/connect-wallet-button.tsx]: Prefer spaces between braces (eg `{ useWallet }`), do you have a strong opinion here?

File: packages/client/src/components/connect-wallet-modal.tsx [hunk 2]
```diff
@@ -62,18 +57,18 @@ function ConnectWalletModal({
                     <div className='connect-wallet-modal-options-container'>
                         <button
                             className='connect-wallet-modal-option'
-                            disabled={!availableProviders.metamask}
+                            // disabled={!availableProviders.metamask}
```
Comment (kkennis) [inline packages/client/src/components/connect-wallet-modal.tsx:60]: What happens if the user doesn't have the Metamask extension installed? I.e. they have no options here?

File: packages/client/src/hooks/use-balance.ts [hunk 3]
```diff
@@ -107,8 +111,10 @@ export const useBalance = ({ pool, wallet }: Props): WalletBalances => {
                     );
                     return allowance;
                 } catch (e) {
+                    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
                     console.error(`Could not get two-sided allowance of contract ${targetAddress} on behalf of ${wallet?.account ?? ''} for token ${tokenAddress}`);
-                    console.error(`Error; ${e.message as string}`);
+                    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
```
Comment (kkennis) [inline packages/client/src/hooks/use-balance.ts]: Lol this file's eslint comments

File: packages/client/src/components/connect-wallet-button.tsx (hunk 1 above)
Comment (luckyOdisetti) [reply packages/client/src/components/connect-wallet-button.tsx]: Nope, i normally just leave it to prettier. Will space it

File: packages/client/src/components/connect-wallet-modal.tsx (hunk 2 above)
Comment (luckyOdisetti) [reply packages/client/src/components/connect-wallet-modal.tsx:60]: We link to install Metamask chrome plugin

File: packages/client/src/hooks/use-balance.ts (hunk 3 above)
Comment (luckyOdisetti) [reply packages/client/src/hooks/use-balance.ts]: 😖

File: packages/client/src/components/connect-wallet-button.tsx (hunk 1 above)
Comment (khcoaching) [reply packages/client/src/components/connect-wallet-button.tsx]: ah should we run prettier on commit?

**Pull request merged**